protocols/bsbport/bsbport_mqtt.c

```diff
@@ -0,0 +1,252 @@
+/*
+ *
+ * Copyright (c) 2015 by Daniel Lindner <daniel.lindner@gmx.de>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+ *
+ * For more information on the GPL, please go to:
+ * http://www.gnu.org/copyleft/gpl.html
+ */
+
+#include "config.h"
+#include "core/debug.h"
+#include "core/bit-macros.h"
+#include "core/util/fixedpoint.h"
+#include "core/util/string_parsing.h"
+#include "protocols/bsbport/bsbport.h"
+#include "protocols/bsbport/bsbport_tx.h"
+#include "protocols/mqtt/mqtt.h"
+
+#define BSBPORT_PUBLISH_FORMAT          BSBPORT_MQTT_TOPIC "/%02x/%02x%02x%02x%02x/%3s"
+#define BSBPORT_SUBSCRIBE_SET_TOPIC     BSBPORT_MQTT_TOPIC "/set/#"
+#define BSBPORT_SUBSCRIBE_QUERY_TOPIC   BSBPORT_MQTT_TOPIC "/query/#"
+#define BSBPORT_SUBSCRIBE_SET_FORMAT    BSBPORT_MQTT_TOPIC "/set/%2hhx/%2hhx%2hhx%2hhx%2hhx/%3s"
+#define BSBPORT_SUBSCRIBE_QUERY_FORMAT  BSBPORT_MQTT_TOPIC "/query/%2hhx/%2hhx%2hhx%2hhx%2hhx"
+#define BSBPORT_MQTT_RETAIN             false
+#define DATA_LENGTH                     8
+#define TOPIC_LENGTH                    (sizeof(BSBPORT_MQTT_TOPIC) + 17)
+
+static const char publish_topic_format[] PROGMEM = BSBPORT_PUBLISH_FORMAT;
+
+#ifdef DEBUG_BSBPORT_MQTT
+#define BSBDEBUG(s, ...) debug_printf("BSB " s "\n", ## __VA_ARGS__)
+#else
+#define BSBDEBUG(...) do { } while(0)
```

Comment:
This kind of code sequence is used many times in Ethersex. I was wondering which old broken compiler requires it?